### PR TITLE
Add one-handed Viterbi layout for gaming

### PR DESCRIPTION
**Pull request by vosechu**
Adds a layout for my one-handed viterbi that I use for playing EVE and other games. 

Any and all feedback would be greatly appreciated!!

## Description

- Adds gaming layout
- Adds half-sized macro

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keebio/viterbi/keymaps/vosechu/keymap.c [hunk 1]
```diff
@@ -0,0 +1,116 @@
+#include QMK_KEYBOARD_H
+#include "vosechu.h"
+
+extern keymap_config_t keymap_config;
```
Comment (drashna) [inline keyboards/keebio/viterbi/keymaps/vosechu/keymap.c:4]: This isn't actually needed, as it's included already.

```suggestion
```

File: keyboards/keebio/viterbi/keymaps/vosechu/keymap.c [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  return state;
+}
+
+void suspend_power_down_user(void) {
+    rgblight_disable();
+}
+
+void suspend_wakeup_init_user(void) {
+    rgblight_enable();
+}
```
Comment (drashna) [inline keyboards/keebio/viterbi/keymaps/vosechu/keymap.c:101]:
```suggestion
```

You can define `RGBLIGHT_SLEEP` to do this, actually.

File: keyboards/keebio/viterbi/keymaps/vosechu/config.h [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+// #define MASTER_RIGHT
+// #define EE_HANDS
+
+#undef RGBLED_NUM
+#define RGBLIGHT_ANIMATIONS
+#define RGBLED_NUM 12
+#define RGBLIGHT_HUE_STEP 8
+#define RGBLIGHT_SAT_STEP 8
+#define RGBLIGHT_VAL_STEP 8
```
Comment (drashna) [inline keyboards/keebio/viterbi/keymaps/vosechu/config.h:32]:
```suggestion
#define RGBLIGHT_VAL_STEP 8
#define RGBLIGHT_SLEEP
```

File: keyboards/keebio/viterbi/keymaps/vosechu/keymap.c (hunk 2 above)
Comment (vosechu) [reply keyboards/keebio/viterbi/keymaps/vosechu/keymap.c:101]: Awesome! Thank you! I'll fix this.

File: keyboards/keebio/viterbi/keymaps/vosechu/keymap.c (hunk 1 above)
Comment (vosechu) [reply keyboards/keebio/viterbi/keymaps/vosechu/keymap.c:4]: Whew! I'm glad. I hadn't seen this before, so I just accepted it as dark voodoo that had to be kept. Really glad you're keeping an eye on PRs!

**Comment by vosechu** [comment]
@drashna All fixed up. Thank you!

File: keyboards/keebio/viterbi/keymaps/vosechu/config.h [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	{ L10, L11, L12, L13, L14, L15, L16 }, \
+	{ L20, L21, L22, L23, L24, L25, L26 }, \
+	{ L30, L31, L32, L33, L34, L35, L36 }, \
+	{ L40, L41, L42, L43, L44, L45, L46 }, \
+	{ KC_NO , KC_NO , KC_NO , KC_NO , KC_NO , KC_NO , KC_NO  }, \
+	{ KC_NO , KC_NO , KC_NO , KC_NO , KC_NO , KC_NO , KC_NO  }, \
+	{ KC_NO , KC_NO , KC_NO , KC_NO , KC_NO , KC_NO , KC_NO  }, \
+	{ KC_NO , KC_NO , KC_NO , KC_NO , KC_NO , KC_NO , KC_NO  }, \
+	{ KC_NO , KC_NO , KC_NO , KC_NO , KC_NO , KC_NO , KC_NO  } \
+}
```
Comment (zvecr) [inline keyboards/keebio/viterbi/keymaps/vosechu/config.h:46]: Safer to base this of an existing layout macro, as it keeps you above the matrix abstraction
```suggestion
#define LAYOUT_ortho_half_5x7( \
    L00, L01, L02, L03, L04, L05, L06, \
    L10, L11, L12, L13, L14, L15, L16, \
    L20, L21, L22, L23, L24, L25, L26, \
    L30, L31, L32, L33, L34, L35, L36, \
    L40, L41, L42, L43, L44, L45, L46 \
    ) \
    LAYOUT_ortho_5x14( \
        L00, L01, L02, L03, L04, L05, L06, KC_NO , KC_NO , KC_NO , KC_NO , KC_NO , KC_NO , KC_NO, \
        L10, L11, L12, L13, L14, L15, L16, KC_NO , KC_NO , KC_NO , KC_NO , KC_NO , KC_NO , KC_NO, \
        L20, L21, L22, L23, L24, L25, L26, KC_NO , KC_NO , KC_NO , KC_NO , KC_NO , KC_NO , KC_NO, \
        L30, L31, L32, L33, L34, L35, L36, KC_NO , KC_NO , KC_NO , KC_NO , KC_NO , KC_NO , KC_NO, \
        L40, L41, L42, L43, L44, L45, L46, KC_NO , KC_NO , KC_NO , KC_NO , KC_NO , KC_NO , KC_NO  \
    )

```

Comment (vosechu) [reply keyboards/keebio/viterbi/keymaps/vosechu/config.h]: Ah! I hadn't realized that was an option! Love this!

File: keyboards/keebio/viterbi/keymaps/vosechu/config.h [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+#undef RGBLED_NUM
+#define RGBLIGHT_ANIMATIONS
+#define RGBLED_NUM 3
+#define RGBLIGHT_HUE_STEP 8
+#define RGBLIGHT_SAT_STEP 8
+
+#define RGBLIGHT_SLEEP // Put the keyboard to sleep when USB goes dark
+
+#define LAYOUT_ortho_half_5x7( \
```
Comment (noroadsleft) [inline keyboards/keebio/viterbi/keymaps/vosechu/config.h:28]: Just a random thought, prompted by this PR, but out of its scope:

I'm not sure I like this naming scheme for defining half of a split keyboard; may be something to discuss moving forward.

Comment (vosechu) [reply keyboards/keebio/viterbi/keymaps/vosechu/config.h:28]: Totally open to suggestions! There's no time pressure to merge this, so happy to change or discuss!

**Comment by zvecr** [comment]
Thanks!

**Pull request merged**